### PR TITLE
Z-index cleanups

### LINKED ISSUE #11155: Refactor: start using css variables for zIndex
We should move zIndexing over to css variables. This will make mass changes easier and also make zIndexing clearer for contributors

**Describe the solution you'd like**

Update the [variables file](https://github.com/forem/forem/blob/master/app/assets/stylesheets/config/_variables.scss) to be something like this.

```css
/////////////
// z index //
/////////////

--zi-app-bar: 100;
--zi-drawer: 300;
--zi-modal: 500;
--zi-snackbar: 700;
--zi-tooltip: 1000;
```

Work through all [existing](https://github.com/forem/forem/search?l=SCSS&q=z-index%3A) z-index references and update it to use a variable.

**Describe alternatives you've considered**

We don't do it

**Additional context**

https://github.com/forem/forem/pull/11132#issuecomment-717860314
https://github.com/forem/forem/pull/11132#discussion_r514088714

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

This PR tries to clean up and unify all our z-indexes. I've introduced several z-indexes as variables as well as utility-first classes:

```
--z-negative: -1; // to hide something behind everything else
--z-elevate: 1; // to elevate something, just a little, in relation to its sibling
--z-sticky: 100; // header, mobile actions bar, generally some sticky bars..
--z-drawer: 200; // sidebar sliding nav drawers
--z-modal: 300; // modals ¯\_(ツ)_/¯
--z-dropdown: 400; // all kinds of dropdowns
--z-popover: 500; // tooltips, snackbars, etc.
```

Utility classes naming is pretty much the same:

```
.z-negative
.z-elevate
...and so on
```

I've decided to go with hundreds (100, 200, ...) values (instead of 1, 2, 3, 4, 5, ...) because we still have lots of random legacy z-indexes in our codebase, thankfully these are usually set between 1-99 so using hundreds is kinda safety net...

This PR also removes some unused CSS I've found along the way.

## Related PRs

- Closes #11155 

## QA Instructions, Screenshots, Recordings

Just click around... It is possible something will be slightly broken since it's pretty tough to test everything. But hopefully not much. We could just roll it out and fix any potential issues if we find any on prod. 

## Added tests?

- [ ] Yes
- [x] No, not needed.
- [ ] I need help with writing tests

## Added to documentation?

- [ ] Docs.forem.com
- [ ] README
- [x] No documentation needed, but I will write forem.dev post about it once it's merged.

### CODE REVIEW

File: app/assets/stylesheets/config/_variables.scss [hunk 1]
```diff
@@ -94,4 +94,16 @@
...
+  // Z-indexes. //////////////////////////////////////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////////////
+
+  --z-negative: -1; // to hide something behind everything else
+  --z-elevate: 1; // to elevate something, just a little, in relation to its sibling
+  --z-sticky: 100; // header, mobile actions bar
+  --z-drawer: 200; // sidebar sliding nav drawers
+  --z-modal: 300; // modals ¯\_(ツ)_/¯
+  --z-dropdown: 400; // all kinds of dropdowns
+  --z-popover: 500; // tooltips, snackbars, etc.
```
Comment (lisasy) [inline app/assets/stylesheets/config/_variables.scss:108]: Nice!

**Review by lisasy — APPROVED** [review]
LGTM!

**Review by fdoxyz — APPROVED** [review]
I like that this actually removed a ton of lines! I took it for a spin around most areas with popovers and stuff like that (sidebars, podcast player, snackbar, dropdowns, etc). It looks good to me :)

File: docs/installation/mac.md [hunk 2]
```diff
@@ -63,7 +63,8 @@ redis-cli ping
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply docs/installation/mac.md:63]: well i didnt do it, but i take the props anyway!

**Pull request merged**